Projects/UOContent/Gumps/AdminGump.cs

```diff
@@ -1579,30 +1580,41 @@ public void AddFirewallHeader()
 
                 for (var i = 0; i < theirAddresses.Length; ++i)
                 {
-                    table.TryAdd(theirAddresses[i], new List<Account> { acct });
+                    // if its an ip that is not in the list, add it
+                    if (!table.TryAdd(theirAddresses[i], new List<Account> { acct }))
```

Comment:
We should do the opposite since this will needlessly create a new list every time.